src/Umbraco.Web/PropertyEditors/ValueConverters/MultiUrlPickerValueConverter.cs

```diff
@@ -15,77 +14,102 @@ namespace Umbraco.Web.PropertyEditors.ValueConverters
 
         public override object ConvertIntermediateToObject(IPublishedElement owner, IPublishedPropertyType propertyType, PropertyCacheLevel referenceCacheLevel, object inter, bool preview)
         {
-            using (_proflog.DebugDuration<MultiUrlPickerValueConverter>($"ConvertPropertyToLinks ({propertyType.DataType.Id})"))
+            using (_profilingLogger.DebugDuration<MultiUrlPickerValueConverter>($"ConvertPropertyToLinks ({propertyType.DataType.Id})"))
             {
+                var value = inter?.ToString();
                 var maxNumber = propertyType.DataType.ConfigurationAs<MultiUrlPickerConfiguration>().MaxNumber;
 
-                if (inter == null)
+                if (string.IsNullOrEmpty(value))
                 {
                     return maxNumber == 1 ? null : Enumerable.Empty<Link>();
                 }
 
+                // Process links
                 var links = new List<Link>();
-                var dtos = JsonConvert.DeserializeObject<IEnumerable<MultiUrlPickerValueEditor.LinkDto>>(inter.ToString());
-
+                var dtos = JsonConvert.DeserializeObject<IEnumerable<MultiUrlPickerValueEditor.LinkDto>>(value);
                 foreach (var dto in dtos)
                 {
+                    var name = dto.Name;
                     var type = LinkType.External;
                     var url = dto.Url;
+                    IPublishedContent content = null;
 
                     if (dto.Udi != null)
                     {
-                        type = dto.Udi.EntityType == Core.Constants.UdiEntityType.Media
-                            ? LinkType.Media
-                            : LinkType.Content;
-
-                        var content = type == LinkType.Media ?
-                             _publishedSnapshotAccessor.PublishedSnapshot.Media.GetById(preview, dto.Udi.Guid) :
-                             _publishedSnapshotAccessor.PublishedSnapshot.Content.GetById(preview, dto.Udi.Guid);
+                        switch (dto.Udi.EntityType)
+                        {
+                            case Core.Constants.UdiEntityType.Document:
+                                type = LinkType.Content;
+                                content = _publishedSnapshotAccessor.PublishedSnapshot.Content.GetById(preview, dto.Udi.Guid);
+                                break;
+                            case Core.Constants.UdiEntityType.Media:
+                                type = LinkType.Media;
+                                content = _publishedSnapshotAccessor.PublishedSnapshot.Media.GetById(preview, dto.Udi.Guid);
+                                break;
+                            default:
+                                // Skip unsupported entity types
+                                continue;
+                        }
 
-                        if (content == null || content.ContentType.ItemType == PublishedItemType.Element)
```

Comment:
How can the content type ever be an element if it's retrieved from the snapshots content/media cache?